### PR TITLE
validate that different instances of InputObjectType with the same name are the same

### PR DESCRIPTION
**Pull request by yanns**
The validation only checks the field names. It cannot catch all cases, but I think it should be sufficient for 99% of the cases.

### CODE REVIEW

File: modules/core/src/main/scala/sangria/schema/Schema.scala [hunk 1]
```diff
@@ -1232,12 +1238,10 @@ case class Schema[Ctx, Val](
               "You can find more info in the docs: http://sangria-graphql.org/learn/#circular-references-and-recursive-types")
         case t: Named if result contains t.name =>
           result.get(t.name) match {
-            case Some(found)
-                if !sameType(found._2, t) && t.isInstanceOf[ScalarAlias[_, _]] && found._2
-                  .isInstanceOf[ScalarType[_]] =>
-              result
             case Some(found) if !sameType(found._2, t) =>
-              typeConflict(t.name, found._2, t, parentInfo)
+              if (t.isInstanceOf[ScalarAlias[_, _]] && found._2.isInstanceOf[ScalarType[_]])
```
Comment (yanns) [inline modules/core/src/main/scala/sangria/schema/Schema.scala:1242]: I've extracted the `if` to avoid calling `sameType` several times.

**Review by xsoufiane — APPROVED** [review]
looks 💯, just a taught maybe you can also add a test to check the case of different field names (not just different classes)

**Review by nikola-mladenovic — APPROVED** [review]
💯

**Comment by yanns** [comment]
> looks 💯, just a taught maybe you can also add a test to check the case of different field names (not just different classes)

Actually, we only test different field names as the class is not persisted with InputObjectType. I'll update the test name.

**Pull request merged**